pkg/tfgen/generate.go

```diff
@@ -850,9 +850,18 @@ func (g *Generator) Generate() error {
 		return errors.Wrapf(err, "failed to create Pulumi schema")
 	}
 
+	// Now push the schema through the rest of the generator.
+	return g.GenerateFromSchema(pulumiPackageSpec)
+}
+
+// GenerateFromSchema creates Pulumi packages from a pulumi schema and the information the
+// generator was initialized with.
+func (g *Generator) GenerateFromSchema(pulumiPackageSpec pschema.PackageSpec) error {
```

Comment:
This new public API is reasonable. Separate PR for this kind of thing is nice for project history (nit).